src/main/java/uk/gov/hmcts/cft/idam/testingsupportapi/service/TestingUserProfileService.java

```diff
@@ -68,6 +86,22 @@ public User createOrUpdateCftUser(String sessionId, User requestUser, String sec
             ));
         }
 
+        if (categories.contains(UserProfileCategory.CASEWORKER)) {
+            Optional<CaseWorkerProfile> existingCaseWorkerProfile = findCaseWorkerProfileForUpdate(
+                idamUser.getId(),
+                idamUser.getEmail()
+            );
+            if (existingCaseWorkerProfile.isEmpty()) {
+                testingCaseWorkerProfileService.createCaseWorkerProfile(sessionId, idamUser);
+            } else if (existingCaseWorkerProfile.get().isSuspended()
```

Comment:
Maybe we want them to be always to be the same and otherwise fail? (right now we only fail if caseworker is suspended and idam user is not)
Also - shall we look at "pending" too?

Reply:
Yeah - so since IDAM is the sole user at the moment we would always expect the entities to be created from scratch. In the event that they aren't being created we do a minimal check that the ids, emails and "status" are all in line, and fail if they aren't. I think we would need to do more with this in the long run, but this is the minimal implementation.